File: backend-spring/domain/src/main/java/apptemplate/domain/entities/Department.java
package apptemplate.domain.entities;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Department entity with business logic.
 */
@Getter
@Setter
@NoArgsConstructor
public class Department extends AuditableEntity {

    private String code;
    private String name;
    private String description;
    private boolean active;

    /**
     * Creates a new department.
     */
    public Department(String code, String name, String description) {
        this.code = code;
        this.name = name;
        this.description = description;
        this.active = true;
    }

    /**
     * Updates department information.
     */
    public void update(String name, String description) {
        this.name = name;
        this.description = description;
    }

    /**
     * Updates department code.
     */
    public void updateCode(String code) {
        this.code = code;
    }

    /**
     * Sets department active status.
     */
    public void setActiveStatus(boolean active) {
        this.active = active;
    }
}

File: backend/dotnet/nlayer-architecture/full/src/App.Template.Api/Models/Entities/Department.cs
namespace App.Template.Api.Models.Entities;

public class Department : AuditableEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;
    public ICollection<User> Users { get; set; } = new List<User>();
}
